specs/ClearBrowsingData.md

```diff
@@ -0,0 +1,289 @@
+  /// are listed below. These data kinds follow a hierarchical structure in which 
+  /// nested bullet points are included in their parent bullet point's data kind.
+  /// Ex: DOM storage is encompassed in site data which is encompassed in the profile data. 
+  /// Each of the following bullets correspond to a COREWEBVIEW2_BROWSING_DATA_KIND. 
+  /// * Profile
+  ///   * Site Data
+  ///     * DOM Storage: App Cache, File Systems, Indexed DB, Local Storage, Web SQL, Cache 
+  ///         Storage
+  ///     * Cookies 
+  ///   * HTTP Cache 
+  ///   * Download History
+  ///   * General Autofill 
+  ///   * Password Autosave
+  ///   * Browsing History
+  ///   * Settings  
+  /// The completed handler will be invoked when the browsing data has been cleared and will 
+  /// indicate if the specified data was properly cleared.
+  /// The first ClearBrowsingData method clears the dataKinds for all time. 
+  /// The second ClearBrowsingData method takes in two additional parameters for the 
+  /// start and end time in which it should clear the data between.  The startTime and endTime 
+  /// parameters correspond to the number of seconds since the UNIX epoch. 
+  HRESULT ClearBrowsingData(
+      [in] uint64 dataKinds,
+      [in] ICoreWebView2ClearBrowsingDataCompletedHandler *handler);
+  HRESULT ClearBrowsingDataInTimeRange(
+      [in] uint64 dataKinds, 
+      [in] double startTime,
+      [in] double endTime, 
+      [in] ICoreWebView2ClearBrowsingDataCompletedHandler *handler);
+}
+
+/// The caller implements this interface to receive the ClearBrowsingData result.
+[uuid(c2b78e49-5bf5-4d38-a535-668a8a8a30d9), object, pointer_default(unique)]
+interface ICoreWebView2ClearBrowsingDataCompletedHandler : IUnknown {
+  /// Provide the completion status and result of the corresponding
+  /// asynchronous method. The result indicates if the ClearBrowsingData 
+  /// call succeeded, was interrupted, or failed. 
+  HRESULT Invoke(
+      [in] HRESULT errorCode, 
+      [in] COREWEBVIEW2_CLEAR_BROWSING_DATA_RESULT_KIND result);
+}
+```
+### .NET, WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    enum CoreWebView2ClearBrowsingDataResultKind
+    {
+        Succeeded = 0,
+        Interrupted = 1, 
+        Failed = 2,
+    };
+
+    enum CoreWebView2BrowsingDataKind
+    {
+        AppCache = 1,
+        FileSystems = 2,
+        IndexedDb = 4,
+        LocalStorage = 8,
+        WebSql = 16,
+        CacheStorage = 32,
+        DomStorage = 63,
+        Cookies = 64,
+        Site = 127,
+        HttpCache = 128,
+        DownloadHistory = 256,
+        GeneralAutofill = 512,
+        PasswordAutosave = 1024,
+        BrowsingHistory = 2048,
+        Settings = 4096,
+        Profile = 8191,
+    };
+
+    public partial class CoreWebView2Environment
+    {
+        public async Task<CoreWebView2ClearBrowsingDataResultKind> ClearBrowsingDataAsync(ulong dataKinds);
```

Comment:
Are we using ulong here because we're worried we can't take the enum because we can take various enum values or'ed together which aren't a part of the enum values? If so, can we add a // comment here asking API reviewers about this? I think but not sure its valid in .NET, maybe valid in WinRT, and not sure about Win32.

Something like
// We're using ulong here because we need to accept or'ed enum values like `HttpCache | DownloadHistory` which isn't a value in the DataKind enum. Is it OK to take the enum as the parameter type here in COM, WinRT, and .NET?

Reply:
We are using ulong here because I needed to cast it to ulong to get the proper mask passed through, and sounds good.

Reply:
Followup: still need to add comments about DPI issues / corner cases.

Reply:
👍